lib/view_component/sidecar_i18n.rb

```diff
@@ -0,0 +1,48 @@
+# frozen_string_literal: true
+
+require "set"
+require "i18n"
+require "action_view/helpers/translation_helper"
+
+module ViewComponent
+  module SidecarI18n
+    def self.included(base)
+      base.class_attribute :i18n_backend, instance_writer: false, instance_predicate: false
+      base.extend ClassMethods
+    end
+
+    class I18nBackend < ::I18n::Backend::Simple
+      def initialize(load_paths)
+        @load_paths = load_paths
+      end
+
+      # Ensure the Simple backend won't load paths from ::I18n.load_path
+      def load_translations
+        super(@load_paths)
+      end
+    end
+
+    def translate(key = nil, locale: nil, **options)
+      locale ||= ::I18n.locale
+
+      catch(:exception) do
+        if key.is_a?(Array)
+          key.map { |k| i18n_backend.translate(locale, k, options) }
+        else
+          i18n_backend.translate(locale, key, options)
+        end
+      end
+    end
+    alias :t :translate
+
+    module ClassMethods
+      def compile(*, **)
```

Comment:
Can we migrate this to use `_after_compile`?

I think it's fine to require the `super` call on it for now since there will be at least a few consumers of that API now.